### PR TITLE
Add vagrant configuration

### PR DESCRIPTION
**Pull request by mdek2053**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,22 +1,50 @@
...
+- ml-lib: [ml-lib](https://github.com/remla24-team6/ml-lib)
+
+## Running via docker-compose:
 
 1. Make sure that docker is installed.
 2. Clone the repo
 3. Navigate into the repo and run `docker-compose up`
 4. Go to [http://0.0.0.0:8000/](http://0.0.0.0:8000/) and test the system.
 
+Note that due to time constraints, the model is not trained extensively. This means that the system will likely give some random-ish output.
```
Comment (Roodster) [inline README.md]: This can be removed.

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+worker_node_mem = 6
+worker_node_ip_base = "192.168.61"
+
+
+Vagrant.configure("2") do |config|
+    config.vm.box = "bento/ubuntu-24.04"
+    config.vm.box_version = "202404.26.0"
+
+    # I know this is bad but I cannot get it to work with SSH when running vagrant up.
+    config.ssh.password = "vagrant"
```
Comment (Roodster) [inline Vagrantfile:16]: Please have another look at setting up ssh. This is indeed bad.

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    config.vm.define "control-node" do |control_node|
+        control_node.vm.network "private_network", ip: control_node_ip
+        control_node.vm.provider "virtualbox" do |vb|
+            vb.memory = control_node_mem * 1024
+            vb.cpus = control_node_cores
+        end
+    end
+    (1..worker_nodes).each do |i|
+        config.vm.define "worker-node-#{i}" do |worker_node|
```
Comment (Roodster) [inline Vagrantfile]: I think this should be defined as node#{i}.

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+
+Vagrant.configure("2") do |config|
+    config.vm.box = "bento/ubuntu-24.04"
+    config.vm.box_version = "202404.26.0"
+
+    # I know this is bad but I cannot get it to work with SSH when running vagrant up.
+    config.ssh.password = "vagrant"
+
+    config.vm.define "control-node" do |control_node|
```
Comment (Roodster) [inline Vagrantfile]: I think this should be defined as "controller".

File: README.md (hunk 1 above)
Comment (mdek2053) [reply README.md]: Removed

File: Vagrantfile (hunk 3 above)
Comment (mdek2053) [reply Vagrantfile]: Done

File: Vagrantfile (hunk 4 above)
Comment (mdek2053) [reply Vagrantfile]: Done

File: Vagrantfile (hunk 2 above)
Comment (mdek2053) [reply Vagrantfile:16]: I have already spent like 5h trying to set up ssh. This way, we can still connect via ssh, it only uses the password during setup. In the assignment description, I cannot find any requirement related to using ssh or passwords on setup so I dont think it is worth it.

Comment (Roodster) [reply Vagrantfile:16]: Ok, I approve

**Review by Roodster — APPROVED** [review]
Please add an issue for the ssh set up. If time permits we can have another look to clean it up.

**Pull request merged**